gammapy/makers/safe.py

```diff
@@ -113,29 +114,46 @@ def make_mask_energy_aeff_default(dataset, observation):
         return mask
 
     def make_mask_energy_aeff_max(self, dataset):
-        """Make safe energy mask from aeff max.
+        """Make safe energy mask from effective area maximum value.
 
         Parameters
         ----------
-        dataset : `~gammapy.spectrum.SpectrumDataset` or `~gammapy.spectrum.SpectrumDatasetOnOff`
+        dataset : `~gammapy.modeling.Dataset`
             Dataset to compute mask for.
 
         Returns
         -------
         mask_safe : `~numpy.ndarray`
             Safe data range mask.
         """
+
         if isinstance(dataset, MapDataset):
-            raise NotImplementedError(
-                "'aeff-max' method currently only supported for spectral datasets"
+            position = self.position
+            if position is None:
+                position = dataset.counts.geom.center_skydir
+            exposure = dataset.exposure
+            position = position.transform_to("icrs")
+            energy = exposure.geom.get_axis_by_name("energy_true")
+            coord = MapCoord.create(
```

Comment:
You can use `MapCoord.create({"skycoord": position, "energy": energy.center})` directly and avoid the coordinate transform before...

Reply:
Done! Thanks